### ISSUE TITLE
vim gn motion is missing

### ISSUE THREAD

**Issue by acornejo** [codemirror/CodeMirror#3851]
A couple of years ago vim introduced the gn motion, which is incredibly useful (especially in conjunction with '.' to repeat actions).

This is not implemented in the editor. For a nice explanation of why the gn motion is so useful, please see here:

http://vimcasts.org/episodes/operating-on-search-matches-using-gn/
https://medium.com/@schtoeffel/you-don-t-need-more-than-one-cursor-in-vim-2c44117d51db

**Comment by marijnh** [comment]
(There's not currently an active maintainer for the vim mode, so issues against it aren't being worked on.)

**Comment by howardjing** [comment]
I'm working on implementing this, still need to clean up my PR / add tests.

**Comment by acornejo** [comment]
Wow, this is super exciting, looking forward to trying it out. Does it work nicely with the '.' repeat operator? I find that gn + '.' are a powerful text editing chord in regular vim/nvim.

**Comment by howardjing** [comment]
Yes, it does work with the `.` operator. I'm currently finding a bunch of bugs with edge cases in my implementation, so not quite ready yet but definitely coming along.

**Comment by marijnh** [comment]
Done in #6413
